trin-history/src/validation.rs

```diff
@@ -0,0 +1,205 @@
+use std::sync::{Arc, RwLock};
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use rlp::Rlp;
+
+use trin_core::{
+    portalnet::types::{content_key::HistoryContentKey, messages::ByteList},
+    types::{
+        header::Header,
+        validation::{HeaderOracle, Validator},
+    },
+};
+
+pub struct ChainHistoryValidator {
+    pub header_oracle: Arc<RwLock<HeaderOracle>>,
+}
+
+#[async_trait]
+impl Validator<HistoryContentKey> for ChainHistoryValidator {
+    async fn validate_content(
+        &mut self,
+        content_key: &HistoryContentKey,
+        content: &ByteList,
+    ) -> anyhow::Result<()>
+    where
+        HistoryContentKey: 'async_trait,
+    {
+        match content_key {
+            HistoryContentKey::BlockHeader(key) => {
+                let rlp = Rlp::new(content);
+                let header = Header::decode_rlp(&rlp)?;
+                let number = format!("0x{:02X}", header.number);
+                let expected_hash = &self
+                    .header_oracle
+                    .write()
+                    .unwrap()
+                    .get_hash_at_height(number)?;
```

Comment:
... continuing from the previous comment, I'd expect to be able to do this here:
```suggestion
                    .get_hash_at_height(header.number)?;
```